aemdesign-aem-common/src/main/content/jcr_root/apps/aemdesign/components/details/generic-details/clientlibs/js/behaviour.js

```diff
@@ -57,19 +50,89 @@ window.AEMDESIGN.components.authoring.genericdetails = AEMDESIGN.components.auth
                         });
                     }
                     else {
-                        //$('.genericdetails-costomfields')[0].items.add();
+                        //add empty row if needed
+                        //$(".pagedetails-pagemetadata")[0].items.add();
                     }
 
 
                 });
 
             } else {
-                console.log(["dialog-genericdetails-behaviour","genericdetails-costomfields","not found"]);
+                console.log(["dialog-"+componentName+"-behaviour","pagemetadata","not found"]);
+            }
+
+
+            //load content into variationName field after fragmentPath is changed
+            if ($("#dialog-"+componentName+"").find("[name='./fragmentPath']")[0]) {
+                var fragmentPath = $("#dialog-"+componentName+"").find("[name='./fragmentPath']")[0];
+
+                console.log(["fragmentPath",fragmentPath,fragmentPath.value]);
+
+
+                //load content into field
+                if ($("#dialog-"+componentName+"").find("[name='./variationName']")[0]) {
+                    var variationName = $("#dialog-"+componentName+"").find("[name='./variationName']")[0];
+                    var variationNamePath = variationName.dataset.fieldPath;
+
+                    console.log(["variationName",variationName,variationNamePath]);
+
+                    $(fragmentPath).on("foundation-field-change", function() {
+
+                        var data = {
+                            fragmentPath: fragmentPath.value
+                        };
+
+                        var url = Granite.HTTP.externalize(variationNamePath) + ".html";
+                        var variationNameRequest = $.get({
+                            url: url,
+                            data: data
+                        });
+
+
+                        // $.getJSON(url).done(function (data) {
+                        //     console.log(["variationName",data]);
+                        // });
+
+                        console.log(["variationNameRequest",variationNameRequest]);
```

Comment:
This will be outputting to the console in prod, right? We should either wrap these debug outputs in some kind of url parameter check (e.g. "?debug=true") or remove them.